core/lib/libtomcrypt/src/tee_ltc_provider.c

```diff
@@ -400,6 +401,39 @@ TEE_Result crypto_hash_get_ctx_size(uint32_t algo, size_t *size)
 	return TEE_SUCCESS;
 }
 
+TEE_Result crypto_hash_alloc_ctx(void **ctx_ret, uint32_t algo)
+{
+	TEE_Result res;
+	size_t ctx_size;
+	void *ctx;
+
+	res = hash_get_ctx_size(algo, &ctx_size);
+	if (res)
+		return res;
```

Comment:
It will be more clear in this:
if (res != TEE_SUCCESS)

Reply:
We've agreed that both forms may be used in the OP-TEE source. This is also in line with the style used in the Linux kernel.